gluten-substrait/src/main/scala/org/apache/gluten/execution/WholeStageTransformer.scala

```diff
@@ -198,6 +208,12 @@ trait TransformSupport extends ValidatablePlan {
 
   // When true, it will not generate relNode, nor will it generate native metrics.
   def isNoop: Boolean = false
+
+  // If set the isCudf, the config will send to native backend,
+  // and enable the cudf plan or not in runtime stage
+  def setIsCudf: Unit = {
+    isCudf = true
+  }
```

Comment:
A flag like this will break Spark's plan immutability / equality? For example, the subquery reuse rule will find a non-cudf node and use it to replace a cudf node since they are actually equal.